zerver/tests/test_auth_backends.py

```diff
@@ -1660,8 +1660,8 @@ def register_extra_endpoints(self, requests_mock: responses.RequestsMock,
 
         self.email_data = email_data
 
-    def get_account_data_dict(self, email: str, name: str) -> Dict[str, Any]:
-        return dict(email=email, name=name)
+    def get_account_data_dict(self, email: str, name: str, user_avatar_url: str='') -> Dict[str, Any]:
+        return dict(email=email, name=name, avatar_url=user_avatar_url)
```

Comment:
this change should probably be in the next commit - in the current commit we don't have `user_avatar_url` yet

Reply:
How could I do this?
Because my solution was:
- Create another branch from master (`fix-commit`)
- Edit `get_account_data_dict`
- Commit changes
- Change to `feature/github-avatar` branch
- Undo changes related to `get_account_data_dict` so changes get acknowledge on next step
- Cherry-pick my commit on the `fix-commit` branch
- Rebase so, I can change the order of the commits and make the change on `fix-commit` come first
- At this point, the commit where I implement the feature doesn't have the changes on `get_account_data_dict`, so I need to redo the changes

That's why the change on `get_account_data_dict` is on both commits.

Reply:
At this point you can simply edit this commit and restore the `get_account_data_dict` functons definition to what it was. (`git rebase -i HEAD^^ -> mark the commit for editing -> change it and finish the rebase` is how I'd do this)

Reply:
Here's how I probably have done at the start (When you had a single commit and you wanted to extract some of these changes into a preliminary one):
1. `git reset --soft HEAD^`
2. git stash
3. Everything has been stashed, so I'd just re-do the changes that are supposed to be in the preliminary commit (they're simple so this should be very quick)
4. Commit these changes as the preliminary commit.
5. `git stash pop` and commit these changes as the main commit again

There are probably better ways, but I'm bad with git :grimacing:

Reply:
Alternatively, since this would have been doable for these changes I think, and much quicker: (VSCode-specific)
1. git reset --soft HEAD^ and unstage everything
2. In VSCode, select the lines with the changes I want in the preliminary commit, right click and "stage selected ranges".
3. Make the prelimiary commit, then stage the remaining changes and make the main commit.

Reply:
hmm I did nothing and now it's correct hahahaha The change on get_account_data_dict is only in the first commit.

Reply:
But thanks for the tips on how to do that :)